nomad/blocked_evals_test.go

```diff
@@ -404,6 +404,12 @@ func TestBlockedEvals_UnblockFailed(t *testing.T) {
 	// Trigger an unblock fail
 	blocked.UnblockFailed()
 
+	// Verify UnblockFailed caused the eval to be immediately unblocked
+	blockedStats := blocked.Stats()
+	if blockedStats.TotalBlocked != 0 && blockedStats.TotalEscaped != 0 {
```

Comment:
@dadgar Let's add an eval with the same job once we have unblocked it, and test whether we track it again.